lfs.c

```diff
@@ -411,6 +411,22 @@ static inline void lfs_superblock_tole32(lfs_superblock_t *superblock) {
     superblock->attr_max    = lfs_tole32(superblock->attr_max);
 }
 
+static inline bool lfs_mlist_isused(struct lfs_mlist *head,
+        struct lfs_mlist *node) {
+    for (struct lfs_mlist **p = &head; *p; p = &(*p)->next) {
+        if (*p == (struct lfs_mlist*)node) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static inline void lfs_mlist_add(lfs_t *lfs, struct lfs_mlist *mlist) {
```

Comment:
If we add this, `lfs_mlist_remove` would be more valuable:
[https://github.com/littlefs-project/littlefs/blob/4c9146ea539f72749d6cc3ea076372a81b12cb11/lfs.c#L2553-L2559 ](https://github.com/littlefs-project/littlefs/blob/4c9146ea539f72749d6cc3ea076372a81b12cb11/lfs.c#L2553-L2559)

Also a nit: I think `lfs_mlist_append` captures the data structure operation better.

Reply:
done both!